File: ObjCSDKVerification/ObjCSDKVerification/FullscreenVC.m
//
//  FullscreenVC.m
//  ObjCSDKVerification
//
//  Created by Siarhei Barmotska on 20/02/2017.
//  All rights reserved.
//

#import "FullscreenVC.h"
#import <AmpiriSDK/AmpiriSDK.h>

@interface FullscreenVC()

@property (strong) AMPFullscreenBannerController *interstitial;
@end

@implementation FullscreenVC

- (void)viewDidLoad {
    [super viewDidLoad];
    __weak typeof(self) wSelf = self;
    [[AmpiriSDK sharedSDK]
     loadFullscreenWithAdUnitId: @"2cb34a73-0012-4264-9526-bde1fce2ba92"
     options: AMPFullscreenLoadOptionsDefault
     forViewController: self
     success: ^(AMPFullscreenBannerController *fullscreenController) {
         __strong typeof(self) sSelf = wSelf;
         sSelf.interstitial = fullscreenController;
         [fullscreenController showFromViewController:sSelf];
     }             failure: ^(AMPError *error) {
         NSLog(@"Error: %@", error.localizedDescription);
     }];
}

@end

File: SwiftSDKVerification/SwiftSDKVerification/FullscreenVC.swift
//
//  FullscreenVC.swift
//  SwiftSDKVerification
//
//  Created by Siarhei Barmotska on 21/02/2017.
//  All rights reserved.
//

import UIKit
import AmpiriSDK

class FullscreenVC: UIViewController {

    fileprivate var interstitial: AMPFullscreenBannerController?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        AmpiriSDK.shared().loadFullscreen(withAdUnitId: "2cb34a73-0012-4264-9526-bde1fce2ba92",
                                          options: .default,
                                          for: self,
                                          success: {[weak self] (fullscreen: AMPFullscreenBannerController?) in
                                            guard let sSelf = self, let myFullscreen = fullscreen else {return}
                                            sSelf.interstitial = myFullscreen
                                            myFullscreen.show(from: sSelf)
                                          },
                                          failure: { (error: AMPError?) in
                                            print("Error : \(error?.localizedDescription)")
                                          })
    }
}
